linera-core/src/remote_node.rs

```diff
@@ -209,9 +210,12 @@ impl<N: ValidatorNode> RemoteNode<N> {
             Vec::new()
         };
 
-        found_blobs.extend(self.try_download_blobs(&missing_blobs).await);
-
-        Ok(found_blobs)
+        if let Some(blobs) = self.try_download_blobs(&missing_blobs).await {
+            found_blobs.extend(blobs);
+            Ok(Some(found_blobs))
+        } else {
+            Ok(None)
+        }
```

Comment:
Previously, if `try_download_blobs` returned nothing, we'd still return `found_blobs`. Right now we return `None`. Is it still correct?

Reply:
`try_download_blobs` only returns `None` if it tried but failed to download at least one blob. In the previous version, that would have caused the lists to have unequal length. (Unless I'm missing something.)